fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -578,11 +577,11 @@ private Query getFilterQuery(Set<ApiFilter> filters) {
                         luceneIndexSearcher,
                         null,
                         query,
-                        perPage,
-                        requestedPageNumber
+                        perPage
                 );
                 hits = hitDocs.scoreDocs;
                 documentCount = hitDocs.totalHits;
+                int requestedPageNumber = paginationParameters.getPage(documentCount);
```

Comment:
As I think about it, I'm not sure this will work.

We'll need to dig into the Lucene library some to be sure, but I'm afraid that the `hitDocs` contains only the first page of data, so `hitDocs.totalHits <= perPage`, not the total number of possible documents.